File: src/etch/bindings/java/util/StrStrHashMap.java
/* $Id$
 *
 * Created by wert on Jun 25, 2007.
 *
 * Copyright (c) 2007 Cisco Systems, Inc. All rights reserved.
 */

package etch.bindings.java.util;

import java.util.HashMap;

/**
 * A HashMap which is from String to String.
 * 
 * To use StrStrHashMap in an etch service, define an extern type and prefix
 * it with this annotation:
 * 
 * @Extern( java, "etch.bindings.java.util.StrStrHashMap", "",
 *   "etch.bindings.java.util.StrStrHashMapSerializer", "" )
 * extern StrStrHashMap
 */
public class StrStrHashMap extends HashMap<String, String>
{
	private static final long serialVersionUID = 6910845292628589771L;

	/**
	 * Constructs a StrStrHashMap initialized from another.
	 * @param other
	 */
	public StrStrHashMap( StrStrHashMap other )
	{
		super( other );
	}
	
	/**
	 * Constructs a StrStrHashMap.
	 */
	public StrStrHashMap()
	{
		// nothing to do.
	}
}

File: src/etch/bindings/csharp/util/StrStrHashMap.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace Etch.Util
{
    /// <summary>
    /// A hashmap which is from string to integer
    /// </summary>
    public class StrStrHashMap : Dictionary<string, string>
    {
        public StrStrHashMap()
        {
            // nothing to do .
        }

        /// <summary>
        /// Constructs a StrStrHashMap initialized from another
        /// </summary>
        /// <param name="other"></param>
        public StrStrHashMap( StrStrHashMap other )
            : base( other )
        { }

    }
}
